src/tye/Program.DefaultOptionsMiddleware.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Linq;
+
+namespace Microsoft.Tye
+{
+    public static partial class Program
+    {
+        private const string DefaultOptionsEnvVarPrefix = "TYE_";
+        private const string DefaultOptionsEnvVarPostfix = "_ARGS";
+
+        private static void DefaultOptionsMiddleware(InvocationContext context)
+        {
+            if (context.ParseResult.CommandResult.Parent != context.ParseResult.RootCommandResult)
+            {
+                return;
+            }
+
+            var commandName = context.ParseResult.CommandResult.Command.Name;
+
+            var rawDefaultOptions = Environment.GetEnvironmentVariable(DefaultOptionsEnvVarPrefix + commandName.ToUpper() + DefaultOptionsEnvVarPostfix);
+            if (string.IsNullOrWhiteSpace(rawDefaultOptions))
+            {
+                return;
+            }
+
+            var originalParseResult = context.ParseResult;
```

Comment:
I think an example here in comments would be useful. Took me a while to understand what was going on here.

Reply:
I added some comments to the code, but here if brief description.

- Line 24 - Get value from env-var for current command - if nothing found then exit
- Line 31 - Get all applied options in original input
- Line 32 - Recreate original input
- L 34 - Check if option `--no-default` is applied in original options - if yes, then exit
Example: `tye run --no-default` - this will not apply default options
- L 39 - Parse default options to validate them
- L 40 - Get valid default options
- L 41 - Filter only the default options which are not already applied in original input
- L 43 - Set new parse result (original input plus default options)